### PR TITLE
Search option for contentblocks picker (searches within category)

### LINKED ISSUE #26: Suggestion: search content blocks
I was just thinking; wouldn't it be nice to have a small search box for the "Add content block" modal window? So you could easily search for a certain content block (by name)?
Maybe even autofocus this field so you can start typing immediately?

### PR DESCRIPTION
**Pull request by bramdoppen**
Search for contentblocks within the selected category. Closes #26 

It's an input value that filters the array, as seen here:
image

**Features**
- Search input has autofocus when you open the overlay;
- When you close this overlay, the input will be cleared (don't know if users prefer to keep their search, but if needed we can change that in the future);
- Search input won't be cleared when changing tabs (when you look for a specific block, it will be there when you click through the tabs, can also change in future).

**Demo**
d8c2ab4d-73f4-4597-b2a8-f18e9d01aa37

### CODE REVIEW

**Comment by PerplexDaniel** [comment]
Very nice! We just need to put the static text in localization files before we can release it (with Google Translate for Danish ;-)).

File: src/Perplex.ContentBlocks/App_Plugins/Perplex.ContentBlocks/perplex.content-blocks.html [hunk 1]
```diff
@@ -66,6 +73,9 @@ <h3>
                             </div>
                         </div>
                     </div>
+                    <div ng-if="filteredResults.length < 1">
+                        <localize key="perplexContentBlocks_noContentBlocksMatchedYourSearch">No Content Blocks matched your search with</localize><span> "{{vm.state.ui.picker.search}}".</span>
+                    </div>
```
Comment (PerplexDaniel) [inline src/Perplex.ContentBlocks/App_Plugins/Perplex.ContentBlocks/perplex.content-blocks.html:78]: This message is now displayed in a grid cell so will not have much space available, is that intended?
msg-grid

I think it's better to move this message to the `<div class="p-col--1-1">` that holds the input and add a bit of `margin-top`:
msg

What do you think?

File: src/Perplex.ContentBlocks/App_Plugins/Perplex.ContentBlocks/perplex.content-blocks.html [hunk 2]
```diff
@@ -66,6 +71,9 @@ <h3>
                             </div>
                         </div>
                     </div>
+                    <div ng-if="filteredResults.length < 1">
+                        No contentblocks matched your search "{{vm.state.ui.picker.search}}".
```
[the remark this answers is not in the document]
Comment (bramdoppen) [reply src/Perplex.ContentBlocks/App_Plugins/Perplex.ContentBlocks/perplex.content-blocks.html]: It's done

File: src/Perplex.ContentBlocks/App_Plugins/Perplex.ContentBlocks/perplex.content-blocks.html (hunk 1 above)
Comment (bramdoppen) [reply src/Perplex.ContentBlocks/App_Plugins/Perplex.ContentBlocks/perplex.content-blocks.html:78]: Agree! I did not test with such a long value ;). Will add it in my next commit.

**Review by PerplexDaniel — APPROVED** [review]

**Pull request merged**